src/quick-lint-js/parse.h

```diff
@@ -3392,6 +3409,26 @@ class parser {
     bool old_value_;
   };
 
+  class depth_guard {
+   public:
+    explicit depth_guard(parser *p, int old_depth_value) noexcept
+        : parser_(p), old_depth_value_(old_depth_value) {
```

Comment:
Instead of accepting `old_depth_value` as a parameter, this constructor could take the value from `p`:
```suggestion
    explicit depth_guard(parser *p) noexcept
        : parser_(p), old_depth_value_(p->depth_) {
```